loopin.go

```diff
@@ -336,34 +334,53 @@ func (s *loopInSwap) executeSwap(globalCtx context.Context) error {
 
 // waitForHtlcConf watches the chain until the htlc confirms.
 func (s *loopInSwap) waitForHtlcConf(globalCtx context.Context) (
-	*chainntnfs.TxConfirmation, error) {
+	*chainntnfs.TxConfirmation, *swap.Htlc, error) {
 
 	ctx, cancel := context.WithCancel(globalCtx)
 	defer cancel()
-	confChan, confErr, err := s.lnd.ChainNotifier.RegisterConfirmationsNtfn(
-		ctx, nil, s.htlc.PkScript, 1, s.InitiationHeight,
+
+	notifier := s.lnd.ChainNotifier
+
+	confChanP2WSH, confErrP2WSH, err := notifier.RegisterConfirmationsNtfn(
+		ctx, nil, s.htlcP2WSH.PkScript, 1, s.InitiationHeight,
 	)
 	if err != nil {
-		return nil, err
+		return nil, nil, err
+	}
+
+	confChanNP2WSH, confErrNP2WSH, err := notifier.RegisterConfirmationsNtfn(
+		ctx, nil, s.htlcNP2WSH.PkScript, 1, s.InitiationHeight,
+	)
```

Comment:
Didn't realize we also need to watch for two address client-side, but makes sense.

Reply:
:+1: